maven-artifact/src/main/java/org/apache/maven/artifact/handler/ArtifactHandler.java

```diff
@@ -41,7 +42,7 @@ public interface ArtifactHandler {
     String getDirectory();
 
     /**
-     * Get the classifier associated to the dependency type.
+     * Returns the classifier of the dependency.
```

Comment:
I'm not sure there is such a thing as a "dependency type" and I'd prefer to avoid that term completely. If it's confusing us, it will certainly confuse end users. I think there are three things in play here:

1. The binary artifact served from the repository, such as a jar file, a signature file, or a pom.xml file 

2. The pom.xml for the artifact. For a jar file this is the pom.xml from which the jar was built, though I'm not sure this is meaningful for anything but the regular jar containing .class files. I'm not sure if it makes sense for javadoc or source jars or non-jars. This is the pom found in the same directory  as the jar.

3. The pom.xml that imports the artifact using a dependency element. 

This dependency can declare a different packaging and other properties of the artifact. 

So does ArtifactHandler report number 3 or number 2 if both are present? Or does the answer to this question depend on which property we're looking at?

Reply:
in fact, if Artifact in Maven core did not mix features from Artifact and Dependency (as Maven Resolver documented and solved 10 years later), this ArtifactHandler should have been named DependencyHandler
Because it's what that class does: define how to handle a dependency
IMHO, saying "artifact/dependency handler" in the javadoc could help understand reality

Reply:
Note that the v4 api provides [`Type`](https://github.com/apache/maven/blob/master/api/maven-api-core/src/main/java/org/apache/maven/api/Type.java) as an equivalent to v3 `ArtifactHandler`, directly accessible using [`Dependency#getType()`](https://github.com/apache/maven/blob/master/api/maven-api-core/src/main/java/org/apache/maven/api/Dependency.java#L25-L31).  Maybe the api could be improved at the same time (replace _artifact type_ with _dependency type_ as a first step for example).